src/Routing/ContentUrlGenerator.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the "StenopePHP/Stenope" bundle.
+ *
+ * @author Thomas Jarrand <thomas.jarrand@gmail.com>
+ */
+
+namespace Stenope\Bundle\Routing;
+
+use Stenope\Bundle\Content;
+use Symfony\Component\Routing\RouterInterface;
+
+class ContentUrlGenerator
+{
+    private RouterInterface $router;
+
+    /** @var RouteInfo[] */
+    private array $routes = [];
+
+    public function __construct(RouterInterface $router)
+    {
+        $this->router = $router;
+    }
+
+    public function generate(Content $content): string
+    {
+        $routeInfo = $this->getRouteInfo($content);
+
+        return $this->router->generate($routeInfo->getName(), [
+            $routeInfo->getMainContentRouteSlugVariable() => $content->getSlug(),
+        ]);
+    }
+
+    private function getRouteInfo(Content $content): RouteInfo
+    {
+        if (!$this->routes) {
+            // Using `$router->getRouteCollection()` is not recommended at runtime.
+            // Might be worth adding a cache warmer to collect the info in cache.
+            $this->routes = RouteInfo::createFromRouteCollection($this->router->getRouteCollection());
+        }
+
+        foreach ($this->routes as $routeInfo) {
+            if ($routeInfo->isMainContentRoute($content->getType())) {
+                return $routeInfo;
+            }
+        }
+
+        throw new \LogicException(sprintf('No main route was defined for type "%s"', $content->getType()));
```

Comment:
So local link without a "main" route results in exception?
Should'nt Stenope leave the link as it is in the content that case?

Reply:
It only throws if it's an actual link to one the content but we don't know how to generate the URL to it. It does not throw on any "local" link.
So throwing in this case properly indicates the user why such relative links between content aren't resolved and what to do next.